boot/freeldr/freeldr/freeldr_i386.lds

```diff
@@ -1,6 +1,6 @@
 SECTIONS
 {
-  .text __image_base__ + __section_alignment__ :
+  .text __image_base__ + __section_alignment__ : ALIGN(__file_alignment__)
```

Comment:
Here's sections of `freeldr_pe.exe` before the change and after.

<details>
<summary>Sections before the change</summary>

![sections_before](https://user-images.githubusercontent.com/16137770/113487148-06485480-94bf-11eb-898d-bb9d3119e942.png)
</details>

<details>
<summary>Sections after the change</summary>

![sections_after](https://user-images.githubusercontent.com/16137770/113487153-0ea08f80-94bf-11eb-8709-4895dad592ac.png)
</details>

Also checked sections of other builds. In all cases had to strip real mode code first. Was doing it with `dd`:
```
dd if=freeldr.sys of=freeldr_pe.exe ibs=512 skip=4
```

<details>
<summary>x86 GCC Debug</summary>

Used `reactos-bootcd-0.4.15-dev-2275-g206156f-x86-gcc-lin-dbg.iso`.

![x86-gcc-debug_sections](https://user-images.githubusercontent.com/16137770/113487536-5de7bf80-94c1-11eb-86da-f9f3551f8015.PNG)

<details>
<summary>x86 GCC Release</summary>

Used `reactos-bootcd-0.4.15-dev-2268-gade5413-x86-gcc-lin-rel.iso`.

![x86-gcc-release_sections](https://user-images.githubusercontent.com/16137770/113487694-32b1a000-94c2-11eb-9716-c79aaade0ad3.PNG)
</details>

<details>
<summary>x86 MSVC Debug</summary>

Used `reactos-bootcd-0.4.15-dev-2268-gade5413-x86-msvc-win-dbg.iso`.

![x86-msvc-debug_sections](https://user-images.githubusercontent.com/16137770/113487796-b9667d00-94c2-11eb-8316-436769de9496.PNG)
</details>

<details>
<summary>x64 MSVC Debug</summary>

Used `reactos-bootcd-0.4.15-dev-2268-gade5413-x64-msvc-win-dbg.iso`.

![x64-msvc-debug_sections](https://user-images.githubusercontent.com/16137770/113487850-182bf680-94c3-11eb-8eec-e0abb4904ada.PNG)
</details>

Reply:
this change bothers me. After checking, it appears that we don't use the GCC linker, but whatever clang default is.

Reply:
Also, GCC produces a file header of size 0x200, whereas clang outputs a fileheader of size 0x1F0

Reply:
I remember gcc invoking `collect2`, but I couldn't make clang use it. Later I will write what do different compilers invoke and figure out what this change relates to.

Reply:
Now we use the GCC linker for executables as well. This was actually a bug in our cmake config. But this doesn't solve the problem, so your change can as well be merged.

Reply:
Can't we use LLD for clang build? What's the problem with it?

Reply:
> 
> 
> Can't we use LLD for clang build? What's the problem with it?

There are entrypoints that are set to 0. LLD allows to set only symbol as an entrypoint, so the linker outputs `lld-link: error: <root>: undefined symbol: _0` while linking `themeui.dll`. I guess it should be done via a linker script then, but I failed to do that either (the linker seems to not accept linker scripts, it just prints `unknown argument: --script=/media/sf_reactos/sdk/cmake/set_entrypoint_to_0.lds` or `lld: error: unknown argument: -T`). Maybe someone knows what to do. Could lld-link be used instead? There's `-flavor link` available for plain lld. I will continue with ld for now.